integrationTests/DonationRepositoryTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace MatchBot\IntegrationTests;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\Charity;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\DonationStatus;
+use MatchBot\Domain\PaymentMethodType;
+
+class DonationRepositoryTest extends IntegrationTest
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->clearPreviousCampaignsCharitiesAndRelated(); // Avoid e.g. HMRC ref dupes
+    }
+
+    public function testItFindsGiftAidSendableDonationsForCharityThatIsReady(): void
+    {
+        $charity = $this->prepareOnboardedCharity(withAgentApproval: true);
+        $donation = $this->prepareDonation($charity);
+
+        $sut = $this->getService(DonationRepository::class);
+        $donationsReady = $sut->findReadyToClaimGiftAid(withResends: false);
+
+        $this->assertEquals([$donation], $donationsReady);
+    }
+
+    public function testItFindsNoGiftAidSendableDonationsForCharityPendingAgentApproval(): void
+    {
+        $charity = $this->prepareOnboardedCharity(withAgentApproval: false);
+        $this->prepareDonation($charity);
+
+        $sut = $this->getService(DonationRepository::class);
+        $donationsReady = $sut->findReadyToClaimGiftAid(withResends: false);
+
+        $this->assertEquals([], $donationsReady);
+    }
+
+    private function prepareOnboardedCharity(bool $withAgentApproval): Charity
+    {
+        $charity = new Charity();
+        $charity->setName('Charity Name');
+        $charity->setHmrcReferenceNumber($withAgentApproval ? 'YY54321' : 'NN54321');
```

Comment:
@bdsl I was trying to fix duplicate record errors, probably prior to doing wider data resets.

Reply:
Right, I think for integration tests randomising is probably the best way to avoid duplicate errors. We're not passing `true` and `false` within the same test run anyway, so I think this isn't needed to avoid duplicates within a test run, and the `clearPreviousCampaignsCharitiesAndRelated` runs to make duplicates not a problem between different tests anyway.

I just tried and the tests pass if I change this to

```suggestion
        $charity->setHmrcReferenceNumber($_specificRefNumberDoesntMatterHere = 'any_ref');
```

Reply:
Yes, the workaround was written prior to introducing that method.